module/mutect2.nf

```diff
@@ -17,67 +17,66 @@ workflow mutect2 {
     contamination_table
 
     main:
-        if (params.tumor_only_mode) {
-            normal_name_ch = Channel.from('NO_NAME')
-        } else {
-            run_GetSampleName_Mutect2(normal_bam.flatten())
-            normal_name_ch = run_GetSampleName_Mutect2.out.name_ch.collect()
-                .map{return (it in List) ? it : [it]}
-        }
-        // to avoid input file name collision or null input error in Mutect2
-        if ( params.multi_tumor_sample ) {
-            contamination_table
-                .flatten()
-                .unique()
-                .filter{ it !== null }
-                .set { contamination_table }
-        }
-
         run_SplitIntervals_GATK(
             params.intersect_regions,
             params.intersect_regions_index,
             params.reference,
             params.reference_index,
             params.reference_dict
-        )
+            )
+
+        Channel
+            .from( params.samples_to_process )
+            .map{ it -> ['orig_id': it['orig_id'], 'id': it['id'], 'sample_type': it['sample_type']] }
+            .set { id_ch }
+
+        normal_orig_ids = id_ch
+            .filter{ it['sample_type'] == 'normal' }
+            .ifEmpty(['orig_id': 'none'])
+            .map{ it['orig_id'] }
+            .collect()
```

Comment:
Rather than mapping to an arbitrary string `none`, I would suggest using the actual built-in `null` value; there's the very rare possibility of a sample being named `none` and using strings like this would cause issues.

Reply:
When `null` is passed as the contamination table path to `mutect2-processes.nf` I get an error: ` Not a valid path value type: org.codehaus.groovy.runtime.NullObject (null)`.  I don't know how to get around this.

Reply:
For the paths, yes you can't pass `null` since it will expect something that looks like a path. I meant here for the IDs, rather than setting to `'none'`, I would use `null` explicitly

Reply:
Oh sorry, I highlighted the wrong one.  If I use `null` for the `normal_orig_ids` then it seems the channel is empty and call_sSNV doesn't run.
` println "normal_orig_ids: ${normal_orig_ids}"` gives
`normal_orig_ids: DataflowVariable(value=null)`

Reply:
Ah I see, in that case, how about a value that's much clearer: `NO_ID`, we typically use that type of pattern, ex. `NO_PATH.bam` for placeholders/filling in undefined values

Reply:
I might quibble with the "much clearer" :-) but sure, I've changed the `none` values to `NO_ID` and `NO_PATH`, and updated testing with `a_mini-mutect2` and `a_mini-mutect2-tumor-only`

Reply:
Fair, I guess it still requires knowing what the intent it but at least it should be consistent now; I don't see the changes here, have you pushed them yet?

Reply:
Indeed, I had forgotten to push the changes.  Done now.